packages/emotion-primitives/src/emotion-native.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import transform from 'css-to-react-native'
+import cssToObjects from 'css-to-object'
+import primitives from 'react-primitives'
+
+import { getStyles, resolveStyles } from './resolveStyles'
+import { splitProps, getPrimitiveProps } from './splitProps'
+
+const StyleSheet = primitives.StyleSheet
+
+/*
+  Renders the styles on iOS, Android and Sketch. Styles are first converted to objects, then to a valid format using StyleSheet.create
+
+  TODO: Pass props to interpolate function
+*/
+function renderStylesOnNative(tag, styles) {
+  // Raw css styles
+  let css
+
+  // Store styles which will be converted to RN
+  const cssToRN = []
+
+  // Store RN styles
+  let rnStyles
+
+  // If styles are not passed using the object styles
+  if (typeof styles === 'object' && Object.keys(styles) == 0) {
+    css = cssToObjects(styles.join(''))
+
+    Object.keys(css).forEach(prop => {
+      cssToRN.push([prop, css[prop]])
+    })
+
+    // Resolve the style using StyleSheet.create
+    rnStyles = resolveStyles([transform(cssToRN)])
+  } else {
+    rnStyles = [...styles]
```

Comment:
**i think** spread in general has some extra work to do (like dealing with array holes, using iteration protocol etc), I guess this is known to be an array so `styles.concat()` should be slightly more performant

Reply:
Yup! We can definitely making changes and add bits related to performance. Since that is emotion's trademark 😉